### ISSUE TITLE
safety(motor): reject non-finite admission scalars and ambiguous action selectors

### ISSUE THREAD

**Issue by Tristan-Stoltz-ERC** [Luminous-Dynamics/symthaea#565]
## Finding — current `main`

The motor/action authority boundary currently assumes floating-point inputs are finite and semantically valid before using ordinary comparisons/casts.

That assumption is not enforced.

## 1. Non-finite Phi fails open under `<` comparisons

`MotorOutputBridge::execute(...)` currently uses:

```rust
if current_phi < min_phi { ... }
...
if current_phi < phi_requirement { ... }
```

and `ActionIR::validate(...)` similarly uses:

```rust
if self.risk_tier() > RiskTier::Low
    && current_phi < policy.capabilities.min_phi
{ ... }
```

plus a shell-specific:

```rust
if current_phi < policy.capabilities.shell.min_phi { ... }
```

For IEEE floating-point `NaN`, ordered comparisons such as `NaN < threshold` are false.

Therefore a non-finite `current_phi` can bypass threshold branches unless another unrelated policy check happens to reject the action.

This is relevant to real effects: `MotorOutputBridge` can be switched to real execution mode, and file writes under an otherwise permitted sandbox/policy path are concrete side effects.

## 2. Non-finite motor confidence also fails open

The motor bridge checks:

```rust
if motor_confidence < MOTOR_CONFIDENCE_MIN { ... }
```

so `NaN` confidence does not enter the low-confidence rejection branch.

## 3. Action selector uses lossy float -> `u8` coercion

Current decoder:

```rust
pub fn from_param(val: f64) -> Option<Self> {
    match val as u8 {
        0 => Some(Read),
        ...
        7 => Some(RunCommand),
        _ => None,
    }
}
```

This does not require an exact finite integer selector.

Depending on Rust float-to-integer cast semantics, fractional/out-of-range/non-finite values are truncated/saturated/coerced before matching. Examples that must be explicitly decided rather than inherited from `as` semantics include:

```text
0.9
1.9
-1.0
NaN
+infinity
```

A malformed numeric signal must never become a valid action merely because the language cast produced an in-range byte.

### Authority scalars

Before any threshold comparison:

```text
Phi/confidence is finite
AND
value is in the documented domain
```

Otherwise:

```text
invalid numeric state -> DENY / no action
```

### Action selector

Before mapping to an action type:

```text
selector is finite
AND
selector satisfies the explicitly documented encoding
```

If the intended encoding is integer IDs, require an exact integer in `[0, 7]`.

If the intended encoding is continuous bins, define those bins explicitly and test every boundary; do not rely on `as u8` truncation/saturation.

### `ActionIR::validate`

Reject non-finite `current_phi` before any risk-specific comparison.

Prefer a dedicated policy error such as:

```text
InvalidAuthorityScalar { field: "current_phi", value }
```

rather than representing NaN as merely "too low" if error compatibility permits.

Also validate policy threshold configuration itself (`min_phi`, shell min Phi) as finite and in `[0,1]` at construction/load boundaries so a NaN threshold cannot disable its own comparison.

### `MotorOutputBridge::execute`

Before action selection or gating, reject:

- non-finite `current_phi`;
- `current_phi` outside the accepted `[0,1]` domain;
- non-finite `motor_confidence`;
- confidence outside the documented `[0,1]` domain;
- invalid/non-finite action selector.

Do not clamp malformed authority inputs into validity at the effect boundary; clamping can hide upstream corruption. Reject and surface evidence.

## PR1 — make selector encoding explicit

Preferred if action IDs are intended:

```rust
pub fn from_param(val: f64) -> Option<Self> {
    if !val.is_finite() || val.fract() != 0.0 || !(0.0..=7.0).contains(&val) {
        return None;
    }
    match val as u8 { ... }
}
```

If the FEP architecture intentionally emits continuous action logits/values, replace this with an explicit decoder whose binning/argmax semantics are named and tested.

### Phi

For at least a real-capable mutation path:

```text
current_phi = NaN -> reject
current_phi = +inf -> reject
current_phi = -inf -> reject
current_phi < 0 -> reject
current_phi > 1 -> reject
```

and verify no executor call/effect occurs.

### Confidence

```text
motor_confidence = NaN/+inf/-inf/out-of-range -> reject
```

### Selector

Explicit tests for:

```text
0.0 -> Read
7.0 -> RunCommand
99.0 -> None
NaN -> None
+inf -> None
-inf -> None
negative finite -> None
fractional values -> exact documented behavior (prefer None for integer encoding)
```

### Policy thresholds

Reject or fail closed on non-finite/out-of-range configured Phi thresholds.

## Broader rule

Authority-bearing floating point must be treated as untrusted structured input:

```text
finite + range/domain validated
    -> compare/classify

raw float
    -> never directly grants authority
```

This should become a reusable review rule across Symthaea's safety, confidence, coherence, and actuator boundaries rather than a one-off NaN patch.

## Interaction with other action issues

- #560 binds authority to the exact post-wisdom action.
- #545 requires explicit effect capabilities.
- #559 makes command risk classification coherent.
- #563 makes primitive target translation fail closed.

This issue ensures the scalar evidence feeding those gates cannot itself be malformed and fail open.

## Non-scope

No judgment about Phi as a scientific measure, no removal of Phi gating, no claim that malformed numeric state is currently attacker-controlled, and no HAL redesign.

The narrow requirement is that **invalid floating-point state cannot be interpreted as sufficient authority**.

Related: #545, #560, #563, #559; Luminous-Dynamics/luminous-dynamics#44.

**Comment by Tristan-Stoltz-ERC** [comment]
Implementation progress on the numeric authority boundary:

- **PR #572** (`bc2f4b26c3ac1e4d3f93e570dab6152ff858c2d1`) — exact finite motor action selector. Rejects NaN/±inf/negative/fractional/out-of-range selector values before float→u8 conversion.
- **PR #577** (`fc0b82d31634bf69be27acefcd4565c01ce3a0a3`) — stacked on #572. Rejects non-finite/out-of-range `current_phi` and `motor_confidence`, plus invalid effective `min_phi`, before any comparison; regressions assert no executor telemetry occurs.

Both remain draft/unqualified while hosted CI is queued.

The **core** `ActionIR::validate(...)` issue remains open and separate: current main still performs raw global and shell-specific `current_phi < threshold` checks. The intended follow-up is to validate `current_phi`, `policy.capabilities.min_phi`, and `policy.capabilities.shell.min_phi` as finite/in-domain before comparison, preferably with a distinct `PolicyViolation::InvalidAuthorityScalar` rather than overloading `PhiTooLow`.

I am deliberately not rewriting the very large `src/action/mod.rs` through the contents API without a narrowly reviewable patch mechanism; the motor-boundary fixes therefore do not close #565 by themselves.
